src/libcore/num/mod.rs

```diff
@@ -2363,7 +2367,9 @@ macro_rules! uint_impl {
         /// ```
         #[stable(feature = "rust1", since = "1.0.0")]
         pub fn checked_next_power_of_two(self) -> Option<Self> {
-            let npot = self.next_power_of_two();
+            let bits = size_of::<Self>() * 8;
+            let one: Self = 1;
+            let npot = one << ((bits - self.wrapping_sub(one).leading_zeros() as usize) % bits);
```

Comment:
Isn't this going to panic if `self == 0`?
We need a check for this condition first, the same way you wrote one in `fn next_power_of_two`.

Also, I'd personally find the following slightly easier to understand (but I don't mind the current code either):
```rust
let shift = bits - self.wrapping_sub(one).leading_zeros() as usize;
if shift < bits {
    Some(one << shift)
} else {
    None
}
```